sonar-dotnet-shared-library/src/main/java/org/sonarsource/dotnet/shared/plugins/DotNetSensor.java

```diff
@@ -109,41 +115,38 @@ private void importResults(SensorContext context) {
   }
 
   /**
-   * If the project does not contain MAIN files OR does not have any found .NET projects (implicitly it has not been scanned with the Scanner for .NET)
+   * If the project does not contain MAIN or TEST files OR does not have any found .NET projects (implicitly it has not been scanned with the Scanner for .NET)
    * we should log a warning to the user, because no files will be analyzed.
    *
-   * @param hasMainFiles True if MAIN files of this sensor language have been indexed. Can be true only if `hasProjects` is false.
-   * @param hasProjects  True if at least one .NET project has been found in {@link org.sonarsource.dotnet.shared.plugins.FileTypeSensor#execute(SensorContext)}. Can be true only if `hasMainFiles` is false.
+   * @param hasFilesOfLanguage True if ANY files of this sensor language have been indexed.
+   * @param hasProjects  True if at least one .NET project has been found in {@link org.sonarsource.dotnet.shared.plugins.FileTypeSensor#execute(SensorContext)}.
    */
-  private void log(FileSystem fs, boolean hasMainFiles, boolean hasProjects) {
-    boolean hasTestFiles = SensorContextUtils.hasFilesOfType(fs, Type.TEST, pluginMetadata.languageKey());
+  private void log(boolean hasFilesOfLanguage, boolean hasProjects) {
     if (hasProjects) {
-      // the scanner for .NET has been used, which means that `hasMainFiles` is false.
-      assert !hasMainFiles;
-      if (hasTestFiles) {
-        warnThatProjectContainsOnlyTestCode(fs, analysisWarnings, pluginMetadata.shortLanguageName());
-      }
-    } else if (hasMainFiles || hasTestFiles) {
-      // the scanner for .NET has _not_ been used. hasMainFiles can be either true or false.
+      // the scanner for .NET has been used, which means that `hasFilesOfLanguage` is false.
+      assert !hasFilesOfLanguage;
+    } else if (hasFilesOfLanguage) {
+      // the scanner for .NET has _not_ been used.
       LOG.warn("Your project contains {} files which cannot be analyzed with the scanner you are using."
           + " To analyze C# or VB.NET, you must use the SonarScanner for .NET 5.x or higher, see https://redirect.sonarsource.com/doc/install-configure-scanner-msbuild.html",
         pluginMetadata.shortLanguageName());
     }
-    if (!hasMainFiles && !hasTestFiles) {
+    if (!hasFilesOfLanguage) {
       logDebugNoFiles();
     }
   }
 
   private static void warnThatProjectContainsOnlyTestCode(FileSystem fs, AnalysisWarnings analysisWarnings, String languageName) {
-    LOG.warn("This {} sensor will be skipped, because the current solution contains only TEST files and no MAIN files. " +
-        "Your SonarQube/SonarCloud project will not have results for {} files. {}",
-      languageName, languageName, READ_MORE_MESSAGE);
+    LOG.warn("The scanner detected only TEST files and no MAIN files in the current solution. " +
+        "Your SonarQube/SonarCloud project will be missing many {} MAIN-code related issues. {}",
```

Comment:
```suggestion
        "Only TEST-code related results will be imported to your SonarQube/SonarCloud project. Many of our rules (e.g. vulnerabilities) are raised only on MAIN code. {}",
```

Reply:
Unfortunately we don't have documentation on this aspect to send the users to (see [internal thread](https://discuss.sonarsource.com/t/rules-sonarsource-com-and-in-product-docs-should-contain-the-rule-scope-main-test/7379/2)), and I think it would be useful to educate them via this message.

Especially as we know that our security offering is evaluated.

Reply:
(and also it's good to keep messages similar between console warnings and GUI warnings)